pandas_datareader/oandarest.py

```diff
@@ -0,0 +1,303 @@
+                    3H  ->  3 hour candlesticks, day alignment
+                    4H  ->  4 hour candlesticks, day alignment
+                    6H  ->  6 hour candlesticks, day alignment
+                    8H  ->  8 hour candlesticks, day alignment
+                    12H -> 12 hour candlesticks, day alignment
+                    1D  ->  1 day candlesticks, day alignment
+                    1W  ->  1 week candlesticks, aligned to start of week
+                    1M  ->  1 month candlesticks, aligned to first day of the month
+                    See OANDA REST v20 for updated list
+            Default: See DEFAULT_FREQUENCY
+        candleFormat: string
+            Candlesticks representation
+            Valid values: M,B,A
+                M for midpoint
+                B for Bid
+                A for Ask
+            Default: See DEFAULT_CANDLE_FORMAT
+        access_credential: Dict of strings
+            Credential to query the api
+            credential["accountType"]="practise". Mandatory. Valid values: practice, live
+            credential["apiToken"]="Your OANDA API token". Mandatory. Valid value: See your OANDA Account's API Token
+    """
+
+    DEFAULT_FREQUENCY = "5S"
+    DEFAULT_CANDLE_FORMAT = "M"
+    SUPPORTED_OFFSET_ALIASES = {
+        "5S": "S5",
+        "10S": "S10",
+        "15S": "S15",
+        "30S": "S30",
+        "T": "M1",
+        "1T": "M1",
+        "2T": "M2",
+        "4T": "M4",
+        "5T": "M5",
+        "15T": "M15",
+        "30T": "M30",
+        "H": "H1",
+        "1H": "H1",
+        "2H": "H2",
+        "3H": "H3",
+        "4H": "H4",
+        "6H": "H6",
+        "8H": "H8",
+        "12H": "H12",
+        "1D": "D",
+        "1W": "W",
+        "1M": "M"
+    }
+
+    def __init__(self, symbols, symbolsTypes=None,
+                 start=None, end=None,
+                 freq=None, candleFormat=None,
+                 session=None,
+                 access_credential=None):
+        _BaseReader.__init__(self, symbols, start=start,
+                             end=end, session=session)
+
+        if symbols is None:
+            self.symbols = ["EUR_USD"]
+
+        self.symbolsTypes = symbolsTypes
+        if symbolsTypes is None:
+            self.symbolsTypes = ["currency"]
+
+        self.freq = freq
+        if freq is None:
+            self.freq = OANDARestHistoricalInstrumentReader.DEFAULT_FREQUENCY
+
+        self.candleFormat = candleFormat
+        if candleFormat is None:
+            self.candleFormat = OANDARestHistoricalInstrumentReader.DEFAULT_CANDLE_FORMAT
+
+        self.access_credential = access_credential
+        if access_credential is None:
+            self.access_credential = {}
+
+        if 'accountType' not in access_credential:
+            self.access_credential['accountType'] = "practice"
+
+        if 'apiToken' not in access_credential:
+            self.access_credential['apiToken'] = ""
+
+    def read(self):
+        dfs = {}
```

Comment:
using an OrderedDict will keep order of symbols in this Panel

    from pandas.compat import OrderedDict